### PR TITLE
[T2A2][T16-A4]Cheng Yong Quan

### PR DESCRIPTION
**Pull request by RiveriaHorizon**

### CODE REVIEW

**Comment by RiveriaHorizon** [comment]
Ready for review.

I can't seem to separate the 2 tutorial exercise(T2A2 and T2A3) from this PR. Send help please!

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -207,14 +206,52 @@
      */
 
     public static void main(String[] args) {
-        showWelcomeMessage();
-        processProgramArgs(args);
-        loadDataFromStorage();
+
+        showToUser(DIVIDER, DIVIDER, VERSION, MESSAGE_WELCOME, DIVIDER);
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:210]: The changes in main() seems intended for A3?

Remember to start a fresh branch from master for each activities in future. :-)

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -794,11 +831,11 @@ private static void addPersonToAddressBook(String[] person) {
      * @return true if the given person was found and deleted in the model
      */
     private static boolean deletePersonFromAddressBook(String[] exactPerson) {
-        final boolean changed = ALL_PERSONS.remove(exactPerson);
-        if (changed) {
+        final boolean isChanged = ALL_PERSONS.remove(exactPerson);
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:834]: A2: good naming change.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -989,12 +1026,12 @@ private static String extractPhoneFromPersonString(String encoded) {
 
         // phone is last arg, target is from prefix to end of string
         if (indexOfPhonePrefix > indexOfEmailPrefix) {
-            return removePrefixSign(encoded.substring(indexOfPhonePrefix, encoded.length()).trim(),
+            return removePrefix(encoded.substring(indexOfPhonePrefix, encoded.length()).trim(),
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1029]: A2: good naming change to remove ambiguous term "sign"

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1144,14 +1181,15 @@ private static String getUsageInfoForExitCommand() {
      */
 
     /**
-     * Removes sign(p/, d/, etc) from parameter string
+     * Removes prefix(p/, d/, etc) from parameter string
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1184]: Correct! Comments need to be updated as well.

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1144,14 +1181,15 @@ private static String getUsageInfoForExitCommand() {
...
-     * @param sign  Parameter sign to be removed
-     * @return  string without the sign
+     * @param stringInput  Parameter as a string
+     * @param prefix  Parameter prefix to be removed
+     * @return  string without the prefix
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefix(String stringInput, String prefix) {
+    	String stringOutput = stringInput.substring(prefix.length());
```
Comment (sooyj) [inline src/seedu/addressbook/AddressBook.java:1191]: This code change is incorrect. This method will always remove the front part of the "stringInput" now. The intention of the method is to remove the prefix **if matched**. 

Hint: Google for replaceFirst().

**Review by sooyj — COMMENTED** [review]
A2 is reasonable, though there seems to be changes intended for A3 as well.

**Pull request closed without merging**